enumerables.rb

```diff
@@ -0,0 +1,143 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield (self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    selected = []
+    my_each { |i| selected << i if yield(i) }
+    selected
+  end
+
+  def my_all?
+    result = true
+    my_each { |i| result = false unless yield(i) }
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each { |i| result = true if yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    my_each { |i| result = false if yield(i) }
+    result
+  end
+
+  def my_count
+    result = 0
+    my_each { |i| result += 1 if yield(i) }
+    result
+  end
+
+  def my_map(arr = nil)
+    result = []
+    if arr.nil?
+      my_each { |i| result.push(yield(i)) }
+    else
+      my_each { |i| result.push result.call(i) }
+    end
+    result
+  end
+
+  def my_inject
+    result = first
+    each_with_index do |item, index|
+      next if index.zero?
+
+      result = yield(result, item)
+    end
+    result
+  end
+
+  def multiply_els
```

Comment:
* [ ] ```multiply_els``` this is not an enumerable method, kindly move it outside the module wrap as required.